docs/html/development/conventions.rst

```diff
@@ -45,27 +45,42 @@ code blocks).
 Indentation
 -----------
 
-We use 4 spaces for indentation.
+We use 3 spaces for indentation.
```

Comment:
Um, why? I'm pretty sure my editor will use 4 spaces by default for most indents, and this is going to be a pain to follow.

Reply:
4 doesn't work with lists, and they become code blocks / literal blocks. :(

3 works with everything. I'm still trying to figure out how to make 4 work though.

Reply:
OK. I don't understand why without doing more experimentation than I have time for right now, but I'm happy to take your word for it. I'd go for "4 except for lists" if we have to, But not 3. In general, though, I dislike mandating things to this level. Why do we need to insist on a specific indentation at all?

To be blunt, I'm probably going to ignore this recommendation anyway. I'll never remember to check the standard, we don't have a linter that complains about violations, and I'm not going to fight my editor over this. I'll probably just stay consistent with surrounding text. I'm happy if someone wants to do occasional "make it fit the standard" passes over the doc source, but I'm not going to get up tight about it myself.

Reply:
I don’t think this is three spaces per se, but following lines need to align relative to the bullet (or number). You’ll need other number of spaces if you go past `10.` or use Roman numerals etc.

I think 4 is still the better choices in general.

Reply:
Yes, [that's the reason](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#bullet-lists).

I hope this is stating the obvious, but even though

```
1.  This is
    some text
```

is apparently acceptable according to the rules, requiring two spaces after the number is not a good way of following a "4 spaces for indentation" rule...

Please, let's just drop the whole recommendation for a specific indent, and go for "must follow ReST rules on indentation/whitespace, and otherwise be consistent with surrounding source text, and readable in source form".

Reply:
I think 3-space indentation is recommended by ReST because this is how the majority of ReST source look like
```rst
.. command:: argument
   more arguments
```
The rare cases where alignment does not matter are indentation of paragraph and nested elements, so for consistency, 3 might be a good choice.  I can (and want to) do the ReST linting CI if that's what's missing here.

Reply:
That fits in with my preference for "make sure it looks readable". I don't think ReST linting is what's missing here - quite the opposite, I'm not a fan of using lint to enforce rules like this which are difficult to agree on (although I'm fine with linting to handle choices we *do* have a good consensus on).

Reply:
Dropped this entire section. :P